### PR TITLE
Fixbug redirect onboard

### PR DESCRIPTION
**Pull request by liadove**
# ⚠️ __IMPORTANT: Please do not create a Pull Request without creating an issue first.__

All changes need to be discusssed before proceeding. Failure to do so may result in the pull request being rejected.

Please be sure to review our [Code of Conduct](https://github.com/COVID-19-electronic-health-system/.github/blob/master/CODE_OF_CONDUCT.md), [Contributing Guidelines](https://github.com/COVID-19-electronic-health-system/.github/blob/master/CONTRIBUTING.md), and [Support documentation](https://github.com/COVID-19-electronic-health-system/.github/blob/master/SUPPORT.md) before submitting a pull request.

-----

## Please include the issue number the pull request fixes by replacing YOUR-ISSUE-HERE in the text below.

Fixes https://github.com/COVID-19-electronic-health-system/Corona-tracker/issues/581

## Summary

Was created a variable showOnboard in the onboarding redux store to indicate if the user didn't fill out onboarding information. The App component will re-render when showOnboard variable will be changed.

## Details

I had an issue to figure out how to re-render the App component because only there we can check if the user filled out onboard info, and when we fetch the data from the backend we needed time to get the user data, so it won't get the data right away and component already would load, so I created showOnboard variable to re-render the App component.

## Final Checklist

- [ ] For CoronaTracker, did you bump the version in `package.json`?
    - Update the second decimal for a major change
    - Update the third decimal for a minor change
    - Numbers can go past 9, e.g. `1.0.9` => `1.0.10`
    - For more info, read about [Semantic Versioning](https://semver.org/)
- [ ] Did you add any new tests as necessary?
- [ ] Is your PR rebased off the most current master?
- [ ] Have you squashed all commits? _(can be done at merge)_
- [ ] Did you use `yarn` not `npm`? _(important!)_
- [ ] Did you use Material-UI wherever possible?
- [ ] Did you run `yarn lint` on the code?
- [ ] Did you run all of your most recent changes locally to make sure everything is working?

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -40,11 +40,14 @@ const App = props => {
...
     document.body.style.zoom = '100%';
-    if (authenticated) {
-      fetchObservations(userSession);
-      fetchDemographicsComorbidities(userSession);
-    }
-  }, [fetchObservations, fetchDemographicsComorbidities, authenticated, userSession]);
+    const fetchData = async function () {
+      if (authenticated) {
+        fetchObservations(userSession);
+        await fetchDemographicsComorbidities(userSession);
```
Comment (acthelemann) [inline client/src/components/App.jsx:46]:
```suggestion
        fetchDemographicsComorbidities(userSession);
```

No need to `await` here

File: client/src/components/App.jsx [hunk 2]
```diff
@@ -40,11 +40,14 @@ const App = props => {
...
+      }
+    };
+    fetchData();
+  }, [fetchObservations, fetchDemographicsComorbidities, authenticated, userSession, showOnboard]);
```
Comment (acthelemann) [inline client/src/components/App.jsx:50]:
```suggestion
  }, [fetchObservations, fetchDemographicsComorbidities, authenticated, userSession]);
```

This `useEffect` isn't dependent on `showOnboard`, so `showOnboard` shouldn't be in the dependencies list

File: client/src/components/OnboardUser.jsx [hunk 3]
```diff
@@ -95,6 +95,10 @@ const OnboardUser = props => {
       [e.target.name]: e.target.value,
     });
   };
+  async function onSubmit() {
+    await setDemographicsComorbiditiesThunk(formState, userSession);
```
Comment (acthelemann) [inline client/src/components/OnboardUser.jsx:99]:
```suggestion
    setDemographicsComorbiditiesThunk(formState, userSession);
```

No need to `await` here

Comment (liadove) [reply client/src/components/OnboardUser.jsx:99]: You're right! Just fixed it and will push, thanks!

**Comment by acthelemann** [comment]
Oh yeah, there's a failing test

**Comment by liadove** [comment]
@acthelemann Yes, I think it's because I added showOnboard in redux. I don't know if I can change the test

**Comment by acthelemann** [comment]
@liadove Since `SET_DEMOGRAPHICS_COMORBIDITIES` now sets `showOnboard` to `!action.formData.age`, you'll have to update the `toEqual` function call in the failing test to pass in the object with `showOnboard` set to `false`.

If you need more explanation on this, let me know

**Comment by liadove** [comment]
@acthelemann Everything passed, thanks!

**Pull request merged**